pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -269,6 +281,20 @@ def get_pixeldata(ds, read_only=False):
             "end of the data"
             .format(actual_length, actual_length - expected_len)
         )
+        # PS 3.3, Annex C.7.6.3
+        if ds.PhotometricInterpretation == 'YBR_FULL_422':
+            # Check to ensure we do have subsampled YBR 422 data
+            ybr_full_length = expected_len / 2 * 3 + expected_len / 2 * 3 % 2
+            # >= as may also include excess padding
+            if actual_length >= ybr_full_length:
+                msg = (
+                    "The Photometric Interpretation of the dataset is "
+                    "YBR_FULL_422, however the length of the pixel data "
+                    "({} bytes) is a third larger than expected ({} bytes) "
+                    "which indicates that this may be incorrect. You may "
+                    "need to change the Photometric Interpretation to "
+                    "the correct value.".format(actual_length, expected_len)
+                )
         warnings.warn(msg)
```

Comment:
Wouldn't you have to skip the resampling below in this case? Or did I misread the code flow?

Reply:
My logic was that that it may all just be excess padding bytes.

Reply:
Ok, maybe in this case expand the warning to state that these extra pixel data will be removed... though that may be a bit excessive, I leave that to your judgement.